onc-healthit/lantern-back-end
# e2e Test Failure Bug

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

Changes in this PR:
The e2e test was passing locally and on github when any of the docker containers failed to boot up. Fixed the Makefile test_e2e and test_e2e_CI (which is used for the github actions e2e test) commands so that they will throw an error when the docker containers fail to start up correctly.

To test:
One way to test this is to make the docker containers fail to start due to not being able to connect to the lantern_mq queue. To do this, remove the line `- ./lanternmq/rabbitmq.conf:/etc/rabbitmq/rabbitmq.conf:ro` from the docker-compose.yml file under the volume section of the lantern_mq container. This will cause the lantern_mq to boot up without the correct lantern username and password, and so the rest of the containers will not be able to connect and will fail and abort, causing the rest of the containers to abort and shut down. Run `make test_e2e`, and make sure that it returns with an error status code.

**Submitter:**
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: [Lantern 351 JIRA Ticket](https://oncprojectracking.healthit.gov/support/browse/LANTERN-351)
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code accomplishes the tasks purpose.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.
